cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/stream/hbase/HBaseStreamConsumerStateStoreFactory.java

```diff
@@ -54,7 +59,9 @@ public synchronized StreamConsumerStateStore create(StreamConfig streamConfig) t
     TableId streamStateStoreTableId = StreamUtils.getStateStoreTableId(namespace);
     TableId hbaseTableId = tableUtil.createHTableId(new NamespaceId(streamStateStoreTableId.getNamespace()),
                                                     streamStateStoreTableId.getTableName());
-    try (HBaseAdmin admin = new HBaseAdmin(hConf)) {
+    try (HBaseDDLExecutor executor = new HBaseDDLExecutorFactory(cConf, hConf).get();
+         // TODO should tableExists be part of HBaseDDLExecutor??
+         HBaseAdmin admin = new HBaseAdmin(hConf)) {
       if (!tableUtil.tableExists(admin, hbaseTableId)) {
```

Comment:
shouldn't have a getAdmin, as that breaks the abstraction. You can probably just always call createTableIfNotExists, unless it's expensive to build the table descriptor. Otherwise, it's probably ok to have the 2 calls, though I would only create the DDLExecutor if the table doesn't exist.